peekaboo/ruleset/rules.py

```diff
@@ -224,6 +224,22 @@ def evaluate(self, sample):
                            True)
 
 
+class OfficeMacroRuleWithAutoAction(Rule):
+    """ A rule checking the sample for Office macros. """
+    rule_name = 'office_macro_with_auto_action'
+
+    def evaluate(self, sample):
+        """ Report the sample as bad if it contains a macro. """
+        if sample.office_macros_with_auto_action:
```

Comment:
It occurs to me that this rule (as well as the classic OfficeMacroRule) doesn't actually decide much. All the actual logic is routed through a method of the Sample class into a function of `toolbox/ms_office.py`. Sample is used purely as a wrapper and cache which results in unnecessary obfuscation and boilterplate code.

Should we now or in a separate step take the opportunity to make this more clear, i.e. similar to cuckoo rules, having the concept of a oletools "analyzer" that is asked by the rule for its opinion and all decision making also done in the rule? This would basically fold `toolbox/ms_office.py` into this class. If this gets too much for the single `rules.py` file we could facter the rules out into separate files.

I'm thinking of making the same change to `sample.submit_to_cuckoo()` mid-term because I don't think it makes much sense for the sample to know how to analyse itself using different analysers but instead the specific rule of the ruleset having exactly this knowledge. While I think it's sensible for the sample to carry the analysis result as a property of itself for later inspection I question if it's good architecture to have it carry logic how to analyse itself as well.

Reply:
Solved well enough for now. Go ahead as is to not engineer it to death.